test/controllers/concerns/hal_actions_test.rb

```diff
@@ -1,29 +1,122 @@
 describe HalActions do
-  class HalActionsTestController < ActionController::Base
+
+  class Foo
+    include ActiveModel::Model
+
+    attr_accessor :id, :is_root_resource, :updated_at
+    cattr_accessor :_id
+
+    self._id = 1
+
+    def self.find(*args)
+      Foo.new.tap { |f|
+        f.id = _id
+        self._id += 1
+        f.updated_at = DateTime.parse('1970-01-01 00:01:00')
+      }
+    end
+
+    def self.inject(*args, &block)
+      self._id = 1
+      [find, find].inject(*args, &block)
+    end
+
+    def self.order(*args); self; end
+    def self.page(*args); self; end
+    def self.per(*args); self; end
+  end
+
+  class FooRepresenter
+  end
+
+  class FoosController < ActionController::Base
     include HalActions
     include Pundit
 
+    cattr_accessor :_caches_action
+
+    attr_accessor :_respond_with
+
     def params
-      { action: 'update' }
+      ActionController::Parameters.new({ action: 'update' })
     end
 
     def current_user
       FactoryGirl.create(:user)
     end
-  end
 
-  let (:controller) { HalActionsTestController.new }
-  let(:account) { create(:account) }
+    def respond_with(*args)
+      self._respond_with = args
+    end
 
-  it 'has cache options' do
-    HalActionsTestController.cache_options.must_equal({compress: true, expires_in: 1.hour, race_condition_ttl: 30})
+    def self.caches_action(action, options={})
+      self._caches_action ||= {}
+      self._caches_action[action] = options
+    end
   end
 
-  describe '#update' do
+  let (:controller) { FoosController.new }
+  let (:account) { create(:account) }
+
+  describe 'instance methods' do
+
+    it 'determines resource id for caching' do
+      controller.show_cache_path.must_equal 60
+    end
+
+    it 'determines the resources id for caching' do
+      controller.index_cache_path.must_equal "c/foos/3-60"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.